app/src/main/java/me/magnum/melonds/ui/shortcutsetup/ShortcutSetupActivity.kt

```diff
@@ -58,14 +46,13 @@ class ShortcutSetupActivity : AppCompatActivity() {
             putExtra(EmulatorActivity.KEY_URI, rom.uri.toString())
         }
         val romIcon = viewModel.getRomIcon(rom)
-        val shortcutInfo = ShortcutInfo.Builder(this, rom.uri.toString())
+        val shortcutInfo = ShortcutInfoCompat.Builder(this, rom.uri.toString())
                 .setShortLabel(rom.name)
-                .setIcon(Icon.createWithBitmap(buildShortcutBitmap(romIcon)))
+                .setIcon(IconCompat.createWithBitmap(buildShortcutBitmap(romIcon)))
```

Comment:
Not sure why this is happening only now, but when clicking on a ROM to create a shortcut, nothing happens and the following error is printed to the Logcat:

`E/JavaBinder: !!! FAILED BINDER TRANSACTION !!!  (parcel size = 1049868)`

Looks like there is a size limit of 1MB in Binder transactions and the size that the icon is taking is... exactly 1MB (512 x 512 x 4). You can read more about this limit here: https://developer.android.com/reference/android/os/TransactionTooLargeException.html

So, one of the fixes here is to reduce the size of the icon's bitmap. Now that I think about it, 512x512 is way to large to be displayed in such a small area. As such, I believe that the size of the icon can be reduced to 256x256 without compromises.

Reply:
Sounds good to me! Will change